fud/fud/executor.py

```diff
@@ -0,0 +1,192 @@
+import time
+
+
+class DummySpinner:
+    """
+    Dummy class to implement the interface of a spinner object.
+    All methods are a sham and do nothing.
+    """
+
+    def __init__(self):
+        pass
+
+    def start(self, text=None):
+        pass
+
+    def stop(self, text=None):
+        pass
+
+    def fail(self, text=None):
+        pass
+
+    def succeed(self, text=None):
+        pass
+
+
+class Profiler:
+    """
+    Interface for profiling runtime
+    """
+
+    def __init__(self):
+        self._current_time = None
+
+    def start(self):
+        assert self._current_time is None, "Attempt to start multiple measurements"
+        self._current_time = time.time()
+
+    def end(self):
+        assert (
+            self._current_time is not None
+        ), "Attempt to end measurement before it starts"
+        t = self._current_time
+        self._current_time = None
+        return time.time() - t
+
+
+class DummyProfiler:
+    """
```

Comment:
@sampsyo is there a more pythonic way to define shims like these? I don't the code for the Executor to have `if self.profiler is not None` everywhere so I did this.

Reply:
I think this is a pretty good route! No shame in this.

I think another good alternative would be to have a bool flag in the `Profiler` class that turns it into a no-op. But this dummy-substitute way seems perfectly reasonable to me!